### PR TITLE
use newer changed attributes accessor

### PR DESCRIPTION
**Pull request by krenzke**
Using the `saved_changes.present?` method here an lead to issues. For example, creating a record and then immediately updating that record will indeed lead to two audits being created. But, the 'update' audit will contain attribute changes for fields that weren't actually updated. Since the update audit creation happens in a `before_update` callback, within that callback the `saved_changes` is still present and refers to the attributes that were saved during the initial record creation.
This PR replaces that logic with was is used in the latest upstream version of this gem.

### CODE REVIEW

File: lib/audited/auditor.rb [hunk 1]
```diff
@@ -185,16 +185,8 @@ def audited_changes
...
-            saved_changes.transform_values(&:first)
-          else
-            changed_attributes
-          end
-          change_list.except(*non_audited_columns).inject({}) do |changes, (attr, old_value)|
-            changes[attr] = [old_value, self[attr]]
-            changes
-          end
+          change_list = respond_to?(:changes_to_save) ? changes_to_save : changes
+          change_list.except(*non_audited_columns)
```
Comment (jmargolis) [inline lib/audited/auditor.rb:189]: It looks like there are other changes in the upstream version https://github.com/collectiveidea/audited/blob/v5.0.2/lib/audited/auditor.rb#L227-L239, are they not needed?

**Review by jmargolis — COMMENTED** [review]
For something like this might want to ask for one of our Rails gurus to review as well.

File: lib/audited/auditor.rb (hunk 1 above)
Comment (krenzke) [reply lib/audited/auditor.rb:189]: I don't believe they are applicable, no. Our version is very far behind the upstream, and things like `redact_values` and `normalize_enum_changes` are new features that were introduced to the audited gem after we forked. As far as I can tell the `non_audited_columns` filtering is the only thing that we do in our fork of the gem that we've tried to keep feature-parity with.

**Review by jmargolis — APPROVED** [review]

**Comment by krenzke** [comment]
Closing this in favor of https://github.com/aha-app/audited/pull/11. It's the same change, just made on the `rails-6` branch, which is now the current one being used in `aha-app`

**Pull request closed without merging**